### PR TITLE
Apply clippy fixes

### PR DESCRIPTION
**Pull request by hecatia-elegua**
I also found a typo / something leading to an error, but just marked it "FIXME" for now, will look into it in another PR.

### CODE REVIEW

**Comment by hecatia-elegua** [comment]
I think [the stride code](https://github.com/danlehmann/bitfield/pull/8/commits/c49f0723b5dd6a40b5f70ee03399924c684f50cc#diff-d835aa021745724d228eab4502c0214643bb2b426c7085d159b06f4955815d65R304) just wants `indexed_count * indexed_stride.unwrap()`.
The problem is that the `NibbleBits64` stride example would calculate:
#[bit(0 ..) -> 16 * 4 + 0 = 64
#[bit(1 (or 2, 3)] 16 * 4 + 1 (or 2, 3) > 64

File: bitbybit-tests/tests/bitfield_tests.rs [hunk 1]
```diff
@@ -637,20 +637,20 @@ fn bitfield_with_u8_enum() {
 
     assert_eq!(
         Ok(NonExhaustiveEnum::Zero),
-        BitfieldWithIndexedEnums::new_with_raw_value(0b00_00000000_000000).val8()
```
Comment (danlehmann) [inline bitbybit-tests/tests/bitfield_tests.rs:640]: These changes are great for the hex values elsewhwere in this file, but for binary the old style worked better as its separators lined up with the bitfield definition

File: bitbybit/src/bitenum.rs [hunk 2]
```diff
@@ -129,24 +129,24 @@ pub fn bitenum(args: TokenStream, input: TokenStream) -> TokenStream {
...
         // this ensures that only constants are being used; due to the way how new_with_raw_value()
         // is written, some expressions would cause compilation issues (e.g. those that refer to other
         // enum values).
-        let int_value = if string_value.starts_with("0x") {
-            u128::from_str_radix(&string_value[2..], 16)
-        } else if string_value.starts_with("0b") {
-            u128::from_str_radix(&string_value[2..], 2)
-        } else if string_value.starts_with("0o") {
-            u128::from_str_radix(&string_value[2..], 8)
+        let int_value = if let Some(stripped) = string_value.strip_prefix("0x") {
```
Comment (danlehmann) [inline bitbybit/src/bitenum.rs:141]: nice, didn't know this syntax!

**Review by danlehmann — APPROVED** [review]
Thanks a lot for the cleanup!

**Comment by danlehmann** [comment]
Is there a way to leave the 0b numbers as-is? If not, let's merge. Fine with me either way

File: bitbybit-tests/tests/bitfield_tests.rs (hunk 1 above)
Comment (hecatia-elegua) [reply bitbybit-tests/tests/bitfield_tests.rs:640]: I noticed that, but thought it was a little gimmicky and doesn't add much readability. If you want it the old way I can turn it back though.

**Comment by hecatia-elegua** [comment]
Then let's just merge.

Is my assumption in https://github.com/danlehmann/bitfield/pull/8#issuecomment-1356872016 correct? I could also move this to an issue.

**Comment by danlehmann** [comment]
> Then let's just merge.
> 
> Is my assumption in [#8 (comment)](https://github.com/danlehmann/bitfield/pull/8#issuecomment-1356872016) correct? I could also move this to an issue.

I don't think so. The generated code looks fine to me:
```rs
        #[bit(0, rw, stride: 4)]
        nibble_bit0: [bool; 16],

        #[bit(1, rw, stride: 4)]
        nibble_bit1: [bool; 16],

        #[bit(2, rw, stride: 4)]
        nibble_bit2: [bool; 16],
```

Turns into the following getters:
```rs
    pub const fn nibble_bit0(&self, index: usize) -> bool { (self.raw_value & (1u64 << (0usize + index * 4usize))) != 0 }
    pub const fn nibble_bit1(&self, index: usize) -> bool { (self.raw_value & (1u64 << (1usize + index * 4usize))) != 0 }
    pub const fn nibble_bit2(&self, index: usize) -> bool { (self.raw_value & (1u64 << (2usize + index * 4usize))) != 0 }
```

**Comment by danlehmann** [comment]
Actually there is an issue: The index isn't bounds-checked.

File: bitbybit-tests/tests/bitfield_tests.rs (hunk 1 above)
Comment (danlehmann) [reply bitbybit-tests/tests/bitfield_tests.rs:640]: Nevermind, this is fine!

**Comment by danlehmann** [comment]
Thanks again for your contribution. This is very much appreciated :)

I filed a task for the missing bounds check: https://github.com/danlehmann/bitfield/issues/10

If you think something else is still broken, please let me know.

**Comment by hecatia-elegua** [comment]
@danlehmann I'm not sure if we're talking about the same thing. I added a "FIXME" somewhere, which also has the right code commented out because it fails a test.
Here: https://github.com/danlehmann/bitfield/pull/8/files#diff-d835aa021745724d228eab4502c0214643bb2b426c7085d159b06f4955815d65L310
The test that fails is `repeated_bitrange_single_bits_with_stride`, which uses `NibbleBits64`.
This fails because of wrong calculation, as shown in my above comment.

I'm currently having a bit of brain fog.

**Pull request merged**